avm/src/vm/opcode.rs

```diff
@@ -2837,15 +2840,25 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
     })
   });
 
-  // IO opcodes
-  io!(waitop => fn(args, hand_mem) {
+  unpred_cpu!(waitop => fn(args, hand_mem) {
     Box::pin(async move {
       let ms = hand_mem.read_fixed(args[0]) as u64;
       sleep(Duration::from_millis(ms)).await;
       hand_mem
     })
   });
 
+  unpred_cpu!(syncop => fn(args, mut hand_mem) {
+    Box::pin(async move {
+      let closure = HandlerFragment::new(args[0], 0);
+      hand_mem.register(CLOSURE_ARG_MEM_START + 1, args[1], true);
```

Comment:
This only works for `execop` and `fetchop` since they only have 1 param right?

Reply:
Mostly correct (`httpsend` also works). This is part of why it's a temporary solution. However, of the `io!` opcodes that aren't the array opcodes that we wouldn't try this on, only the datastore opcodes are incompatible with this, and since we both don't document datastore in docs.alan-lang.org and because we intend to make some changes to it, I figured it was okay to leave that as is for now.